### DISCUSSION TITLE
Keda-operator cannot create KafkaScaledObject due to Kafka client error "client has run out of available brokers to talk to (Is your cluster reachable?)"

### DISCUSSION THREAD

**Discussion by yeskiran** [kedacore/keda#1764] [Q&A / Need Help] [answered]
Hi experts,
I'm trying to setup HPA on my Azure AKS cluster using Keda with Kafka, I'm getting the below error 
" error creating Kafka client: kafka: the client has run out of available brokers to talk to (Is your cluster reachable?)"
Keda version: keda:2.2.0
AKS version : 1.19.7
 Kafka brokers verion 10

My other consumer pods are able to successfully publish and consume messages with the Kafka cluster brokers.

Not sure how to debug this? Only notable difference in my cluster is that Kafka brokers are on 9094 instead of port 9092. Not sure if this the reason why it's failing.

Any help is appreciated.

Thanks
Kiran

**Comment by zroubalik** [comment]
Could you please share your `ScaledObject` spec?

**Reply by yeskiran** [reply]
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: kafka-scaledobject
  namespace: cde
spec:
  scaleTargetRef:
    name: cde-order-manager
  pollingInterval: 30
  triggers:
  - type: kafka
    metadata:
      bootstrapServers: rvkyropkhfka.healthnation.local:9094,rvkyropkhfka02.healthnation.local:9094,rvkyropkhfka03.healthnation.local:9094      
      consumerGroup: CDE.OrderManager.Consumer-1       # Make sure that this consumer group name is the same one as the one that is consuming topics
      topic: cdeaas-az.order-manager.accepted-targets-keda
      # Optional
      lagThreshold: "100"
      offsetResetPolicy: earliest

**Reply by yeskiran** [reply]
![image](https://user-images.githubusercontent.com/29411649/116763299-55fd3b80-a9eb-11eb-8d3e-dbdacfbe2f72.png)

**Comment by yeskiran** [comment, marked as answer]
Hello, 
After lot of troubleshooting, found that the config version property in the sarama client needs to be 0.10.2.0 and only then the client is able to talk to the brokers. I was able to build a sample go app with the same version of sarama client. I just tweaked the version and it was able to connect to kafka brokers (v.0.8) . Is there a way in keda to pass this value as a environment variable?


![kedaSaaramClientCode](https://user-images.githubusercontent.com/29411649/117354222-174e0200-ae7f-11eb-9f47-768ef75e2f8b.png)

Thanks

**Comment by Dorn-** [comment]
Hello, I coded a feature to be able to specify your kafka version.
I'll test it soon and submit a PR asap.

**Reply by deveshbajaj59** [reply]
Any update on this issue.

I am trying the same to connect kafka via sasl_scram_sha512 and getting the same error

**Reply by Dorn-** [reply]
https://github.com/kedacore/keda/pull/1866 has been merged, waiting for the new release of Keda (2.4.0 I guess)
